### PR TITLE
WELD-2197 Replace Guava - basic utilities

### PR DESCRIPTION
**Pull request by mkouba**

### CODE REVIEW

File: impl/src/main/java/org/jboss/weld/util/Defaults.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        // TODO Use the appropriate immutable map replacement (and possibly builder)
+        JLS_PRIMITIVE_DEFAULT_VALUES = ImmutableMap.copyOf(values);
+    }
+
+    private static <T> void put(Map<Class<?>, Object> map, Class<T> type, T value) {
+        map.put(type, value);
+    }
+
+    /**
+     * See also JLS8, 4.12.5 Initial Values of Variables.
```
Comment (manovotn) [inline impl/src/main/java/org/jboss/weld/util/Defaults.java]: `See also JLS8,` -> `JLS7`
I checked them both and they are identical as far as impl goes, but we should probably avoid mentioning J8.

File: impl/src/test/java/org/jboss/weld/tests/unit/util/collections/WeldCollectionsTest.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2014, Red Hat, Inc., and individual contributors
```
Comment (manovotn) [inline impl/src/test/java/org/jboss/weld/tests/unit/util/collections/WeldCollectionsTest.java]: `2014` -> `2016` (5 occurences in total)
Just to keep things up to date :)

File: impl/src/main/java/org/jboss/weld/util/Defaults.java (hunk 1 above)
Comment (mkouba) [inline impl/src/main/java/org/jboss/weld/util/Defaults.java]: I don't think it's a problem. I believe it's the same for all java versions.

File: impl/src/test/java/org/jboss/weld/tests/unit/util/collections/WeldCollectionsTest.java (hunk 2 above)
Comment (mkouba) [inline impl/src/test/java/org/jboss/weld/tests/unit/util/collections/WeldCollectionsTest.java]: This file was created in 2014 so I think it's ok to leave it as it is ;-)

**Comment by mkouba** [comment]
Merged.

**Comment by mkouba** [comment]
retest this please

**Pull request closed without merging**